bin/lib/Logos/Generator/Base/Property.pm

```diff
@@ -27,47 +41,89 @@ sub definition {
-	}
+	if (!$readonly) {
+		# Build getter
+		my $getter_func = "__attribute__((used)) "; # If the property is never accessed, clang's optimizer will remove the getter/setter if this attribute isn't specified
+		$getter_func .= "static $propertyType $propertyGetterName($propertyClass * __unused self, SEL __unused _cmd) ";
+		if($wrapValue) {
+			$getter_func .= "{ NSValue * value = objc_getAssociatedObject(self, (void *)$propertyGetterName); $propertyType rawValue; [value getValue:&rawValue]; return rawValue; }";
+		} else {
+			$getter_func .= "{ return ($propertyType)objc_getAssociatedObject(self, (void *)$propertyGetterName); }";
+		}
 
-	# Build setter
-	my $setter_func = "__attribute__((used)) "; # If the property is never accessed, clang's optimizer will remove the getter/setter if this attribute isn't specified
-	$setter_func .= "static void $propertySetterName($propertyClass * __unused self, SEL __unused _cmd, $propertyType rawValue) ";
-	if($wrapValue) {
-		$setter_func .= "{ NSValue * value = [NSValue valueWithBytes:&rawValue objCType:\@encode($propertyType)]; objc_setAssociatedObject(self, (void *)$propertyGetterName, value, OBJC_ASSOCIATION_RETAIN_NONATOMIC); }";
-	} else {
-		$setter_func .= "{ objc_setAssociatedObject(self, (void *)$propertyGetterName, rawValue, $propertyAssociationPolicy); }";
+		# Build setter
+		my $setter_func = "__attribute__((used)) "; # If the property is never accessed, clang's optimizer will remove the getter/setter if this attribute isn't specified
+		$setter_func .= "static void $propertySetterName($propertyClass * __unused self, SEL __unused _cmd, $propertyType rawValue) ";
+		if($wrapValue) {
+			$setter_func .= "{ NSValue * value = [NSValue valueWithBytes:&rawValue objCType:\@encode($propertyType)]; objc_setAssociatedObject(self, (void *)$propertyGetterName, value, OBJC_ASSOCIATION_RETAIN_NONATOMIC); }";
+		} else {
+			$setter_func .= "{ objc_setAssociatedObject(self, (void *)$propertyGetterName, rawValue, $propertyAssociationPolicy); }";
+		}
+
+		return "$getter_func;\n$setter_func";
 	}
 
-	return "$getter_func; $setter_func";
+	return "";
 }
 
 sub initializers {
 	my $self = shift;
 	my $property = shift;
 
-	my $className = Logos::sigil("class", $property->group->name, $property->class->name);
+	my $className = $property->class->name;
+	my $logosClassVar = Logos::sigil("class", $property->group->name, $className);
+	my $readonly = $property->readonly;
+	my $retain = $property->retainFlag;
+	my $copy = $property->copyFlag;
+	my $nonatomic = $property->nonatomicFlag;
+	my $name = $property->name;
 	my $propertyType = $property->type;
 	my $propertyGetter = $property->getter;
 	my $propertyGetterName = $self->getterName($property);
 	my $propertySetter = $property->setter;
 	my $propertySetterName = $self->setterName($property);
 
-	my $build = "{ char _typeEncoding[1024];";
+	$propertyType =~ /([^ \*]+).+/;
+	my $propertyClassName = $1;
```

Comment:
Will this work if I declare a property with generics (`NSArray<NSString *> *`)? Does that actually work currently 🤔?

Reply:
Oop, I didn't test that. It works for `T *` currently.

Reply:
Fixed by adding `<` to the `[^…]` part. This regex assumes the property type is captured property already, so it just matches the first blob that doesn't contain a character that can't be part of a valid identifier, like ` ` or `*` or `<`

`^([^ \*<]+).+`